### PR TITLE
Delete ReflectionModule::m_pCreatingAssembly and PEAssembly::m_creator

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs [hunk 1]
```diff
@@ -1,29 +1,13 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-// For each dynamic assembly there will be two AssemblyBuilder objects: the "internal"
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs:4]: Obsolete comment

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs [hunk 2]
```diff
@@ -87,6 +71,16 @@ internal ModuleBuilder GetModuleBuilder(RuntimeModule module)
             {
                 throw new ArgumentException(SR.Format(SR.Arg_EnumIllegalVal, (int)access), nameof(access));
             }
+            if (callingAssembly == null)
+            {
+                // Called either from interop or async delegate invocation. Rejecting because we don't
+                // know how to set the correct permission on the new dynamic assembly.
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs:77]:
```suggestion
                // know how to set the correct context of the new dynamic assembly.
```

**Comment by jkotas** [comment]
The test failure is #61359

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs [hunk 3]
```diff
@@ -150,18 +143,12 @@ private void InitManifestModule()
 
         #region DefineDynamicAssembly
 
-        /// <summary>
-        /// If an AssemblyName has a public key specified, the assembly is assumed
-        /// to have a strong name and a hash will be computed when the assembly
-        /// is saved.
-        /// </summary>
         [DynamicSecurityMethod] // Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod.
```
Comment (VSadov) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs:146]: `DynamicSecurityMethod` no longer needed here?

Comment (jkotas) [reply src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs:146]: It is needed to make `Assembly.GetCallingAssembly` reliable. I will update the comment.

Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs:146]:
```suggestion
        [DynamicSecurityMethod] // Required to make Assembly.GetCallingAssembly reliable
```

File: src/coreclr/vm/peassembly.cpp [hunk 4]
```diff
@@ -869,46 +860,14 @@ PEAssembly *PEAssembly::Create(PEAssembly *pParentAssembly,
...
-            return SString::Empty();
-    }
-
-    return pPEAssembly->m_PEImage->GetPath();
-}
-
-
 // Codebase is the fusion codebase or path for the assembly.  It is in URL format.
 // Note this may be obtained from the parent PEAssembly if we don't have a path or fusion
 // assembly.
```
Comment (VSadov) [inline src/coreclr/vm/peassembly.cpp:873]: Are we slightly changing the behavior here? We will now only return codebase for assemblies loaded from individual files. Maybe need to update the comment.

**Review by VSadov — APPROVED** [review]
LGTM, nice!

File: src/coreclr/vm/peassembly.cpp (hunk 4 above)
Comment (jkotas) [reply src/coreclr/vm/peassembly.cpp]: I have updated the comment here and also fixed the error message for dynamic assemblies. The exception type was right so the tests were passing, but the error message was bogus.

**Pull request merged**